core/lib/zksync_core/src/state_keeper/batch_executor/mod.rs

```diff
@@ -330,7 +342,13 @@ impl BatchExecutor {
                     } else {
                         None
                     };
-                    resp.send((vm_block_result, witness_block_state)).unwrap();
+                    let block_tip_metrics = Self::get_execution_metrics(
+                        None,
+                        &vm_block_result.block_tip_execution_result,
+                    );
```

Comment:
Dumb question: Can't the equivalent of this method be called on `vm_block_result` on the caller side (i.e., in `MiniblockUpdates::extend_from_fictive_transaction()`, so that it's not necessary to pass an additional value everywhere? E.g., rework this method as a crate-public function in the `gas_tracker` module, or as a `ExecutionMetricsForCriteria` constuctor. Or do we plan to change `block_tip_metrics` definition in the future, so that it won't be derivable from `vm_block_result` alone?

Reply:
Definitely makes sense to not pass it everywhere and can be derived at the end. I was trying to make it the same way as we do for non-fictive transactions. But as far as I see the reason why it's so for non-fictive ones is that deriving metrics for them requires `Transaction` object and we don't want to clone it before sending to batch executor